src/network/networkd-dhcp4.c

```diff
@@ -899,6 +916,72 @@ static int dhcp_lease_renew(sd_dhcp_client *client, Link *link) {
         return dhcp4_update_address(link, false);
 }
 
+static int dhcp_mtu_reset_loop_detector_handler(sd_event_source *s, uint64_t usec, void *userdata) {
+        return 0;
+}
+
+static int dhcp_set_mtu(Link *link) {
+        uint16_t mtu;
+        int r;
+
+        assert(link);
+        assert(link->manager);
+        assert(link->manager->event);
+        assert(link->network);
+        assert(link->dhcp_lease);
+
+        if (!link->network->dhcp_use_mtu)
+                return 0;
+
+        r = dhcp_detect_mtu_reset_loop(link);
+        if (r < 0)
+                return r;
+
+        if (link->dhcp4_mtu_reset_loop_detected)
+                return 0;
+
+        r = sd_dhcp_lease_get_mtu(link->dhcp_lease, &mtu);
+        if (r == -ENODATA)
+                return 0;
+        if (r < 0)
+                return r;
+
+        r = link_set_mtu(link, mtu);
+        if (r <= 0)
+                return r;
+
+        if (!link->dhcp4_mtu_reset_loop_detector) {
+                r = sd_event_add_time_relative(
+                                link->manager->event,
+                                &link->dhcp4_mtu_reset_loop_detector,
+                                CLOCK_MONOTONIC,
+                                10 * USEC_PER_SEC,
+                                0,
+                                dhcp_mtu_reset_loop_detector_handler,
+                                link);
+                if (r < 0)
+                        return r;
+
+                r = sd_event_source_set_destroy_callback(
+                                link->dhcp4_mtu_reset_loop_detector,
+                                (sd_event_destroy_t) link_destroy_callback);
+                if (r < 0)
+                        return r;
+
+                link_ref(link);
+        } else {
+                r = sd_event_source_set_time_relative(link->dhcp4_mtu_reset_loop_detector, 10 * USEC_PER_SEC);
+                if (r < 0)
+                        return r;
+
+                r = sd_event_source_set_enabled(link->dhcp4_mtu_reset_loop_detector, SD_EVENT_ONESHOT);
+                if (r < 0)
+                        return r;
+        }
```

Comment:
Hmm, why is necessary to use an event source for this? dhcp_mtu_reset_loop_detector_handler() doesn't actually *do* anything. Why not just store the timestamp of the last time mtu was set, and compare it with `now()` in `dhcp_set_mtu()`?